src/dsp_tools/commands/validate_data/check_duplicate_files.py

```diff
@@ -0,0 +1,76 @@
+from collections import defaultdict
+
+from dsp_tools.cli.args import ValidateDataConfig
+from dsp_tools.commands.validate_data.constants import MAXIMUM_DUPLICATE_FILE_PATHS
+from dsp_tools.commands.validate_data.models.validation import DuplicateFileResult
+from dsp_tools.utils.xml_parsing.models.parsed_resource import ParsedResource
+
+
+def check_for_duplicate_files(
+    parsed_resources: list[ParsedResource], config: ValidateDataConfig
+) -> DuplicateFileResult:
+    """
+    Too many duplicate filepaths in the data may cause the SHACL validator to crash.
+    If one file is referenced n times, this produces n * (n-1) validation errors.
+    Therefore, this programmatic pre-validation prevents crashes during later validation steps.
+
+    Args:
+        parsed_resources: Resources to check
+        config: Validate data config
+
+    Returns:
+        Results for the user and decisions how the program should continue
+    """
+    count_dict = _get_filepath_with_more_than_one_usage(parsed_resources)
+    duplicate_files_must_be_ignored = _determine_if_duplicate_files_must_be_ignored(count_dict)
+    if not duplicate_files_must_be_ignored:
+        return DuplicateFileResult(
+            user_msg=None,
+            duplicate_files_must_be_ignored=False,
+            should_continue=True,
+        )
+    msg = _get_duplicate_msg(count_dict, config.is_on_prod_server)
+    should_continue = True
+    if config.is_on_prod_server:
+        should_continue = False
+    return DuplicateFileResult(
+        user_msg=msg,
+        duplicate_files_must_be_ignored=True,
+        should_continue=should_continue,
+    )
```

Comment:
I find this function a bit misleading:  
Of 16 lines of code, only one actually checks if there are duplicates. The remaining 15 are used to determine how to continue.  
At the same time, there is a function called `_determine_if_duplicate_files_must_be_ignored` which actually only checks the counts.

I would either rename this function to something like `too_many_errors` if you want it to only do the counting. Or I would leave the name, but make it return the entire `DuplicateFileResult`.

Also, 
```python
    should_continue = True
    if config.is_on_prod_server:
        should_continue = False
```
could be simplified to
```python
    should_continue = not config.is_on_prod_server
```
which I think is even easier to read.